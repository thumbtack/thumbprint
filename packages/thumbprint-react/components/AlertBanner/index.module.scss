@import '~@thumbtack/thumbprint-tokens/dist/scss/_index';
@import '~@thumbtack/thumbprint-scss/mixins';

.root {
    display: flex;
    justify-content: center;
    align-items: center;
    padding: $tp-space__3;
    color: $tp-color__black;
    background-color: $tp-color__gray-200;
    font-size: $tp-font__body__2__size;
    line-height: $tp-font__body__2__line-height;

    a {
        font-weight: $tp-font-weight__bold;
        text-decoration: underline;
        color: $tp-color__black;
    }

    @include tp-respond-above($tp-breakpoint__small) {
        padding: $tp-space__4;
        font-size: $tp-font__body__1__size;
        line-height: $tp-font__body__1__line-height;
    }
}

.info {
    background-color: $tp-color__blue-100;
    color: $tp-color__blue-600;

    a {
        color: $tp-color__blue-600;
    }
}

.warning {
    background-color: $tp-color__red-100;
    color: $tp-color__red-600;

    a {
        color: $tp-color__red-600;
    }
}

.caution {
    background-color: $tp-color__yellow-100;
    color: $tp-color__yellow-600;

    a {
        color: $tp-color__yellow-600;
    }
}

.icon {
    flex-shrink: 0;
    margin-right: $tp-space__3;

    @include tp-respond-above($tp-breakpoint__small) {
        margin-right: $tp-space__4;
    }
}
