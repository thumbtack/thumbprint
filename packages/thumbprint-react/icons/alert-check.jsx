import React from 'react';

const AlertCheck = props => (
    <svg {...props}>
        <path
            fillRule="evenodd"
            d="M21.5994,6.2001 C21.1604,5.8681 20.5324,5.9591 20.2014,6.4001 L11.8204,17.5751 L7.6244,14.2191 C7.1944,13.8741 6.5654,13.9451 6.2184,14.3751 C5.8734,14.8061 5.9454,15.4361 6.3744,15.7811 L12.1804,20.4251 L21.7994,7.6001 C22.1314,7.1581 22.0414,6.5311 21.5994,6.2001"
        />
    </svg>
);

AlertCheck.defaultProps = {
    xmlns: 'http://www.w3.org/2000/svg',
    fill: 'currentColor',
    width: '28',
    height: '28',
    viewBox: '0 0 28 28',
};

export default AlertCheck;
