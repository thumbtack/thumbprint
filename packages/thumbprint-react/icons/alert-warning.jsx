import React from 'react';

const AlertWarning = props => (
    <svg {...props}>
        <path
            fillRule="evenodd"
            d="M13.9995633,19 C13.4475633,19 12.9995633,19.448 12.9995633,20 C12.9995633,20.552 13.4475633,21 13.9995633,21 C14.5515633,21 14.9995633,20.552 14.9995633,20 C14.9995633,19.448 14.5515633,19 13.9995633,19 M13.9995633,9 C13.4475633,9 12.9995633,9.448 12.9995633,10 L12.9995633,15.997 C12.9995633,16.549 13.4475633,16.997 13.9995633,16.997 C14.5515633,16.997 14.9995633,16.549 14.9995633,15.997 L14.9995633,10 C14.9995633,9.448 14.5515633,9 13.9995633,9 M23.6805633,21.827 C23.2555633,22.562 22.4925633,23 21.6395633,23 L6.36056325,23 C5.50756325,23 4.74456325,22.562 4.31956325,21.827 C3.89756325,21.099 3.89556325,20.228 4.31456325,19.498 L11.9535633,6.182 C12.3785633,5.442 13.1435633,5 13.9995633,5 C14.8565633,5 15.6215633,5.442 16.0465633,6.182 L23.6855633,19.498 C24.1035633,20.228 24.1025633,21.099 23.6805633,21.827 M25.4195633,18.503 L17.7805633,5.187 C16.9945633,3.818 15.5815633,3 13.9995633,3 C12.4165633,3 11.0045633,3.818 10.2195633,5.187 L2.57956325,18.503 C1.80156325,19.859 1.80556325,21.476 2.58856325,22.829 C3.37456325,24.188 4.78556325,25 6.36056325,25 L21.6395633,25 C23.2135633,25 24.6235633,24.188 25.4115633,22.829 C26.1945633,21.476 26.1985633,19.859 25.4195633,18.503"
        />
    </svg>
);

AlertWarning.defaultProps = {
    xmlns: 'http://www.w3.org/2000/svg',
    fill: 'currentColor',
    width: '28',
    height: '28',
    viewBox: '0 0 28 28',
};

export default AlertWarning;
