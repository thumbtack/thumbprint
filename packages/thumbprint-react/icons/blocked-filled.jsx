import React from 'react';

const BlockedFilled = props => (
    <svg {...props}>
        <path d="M13.9996 2.00006C20.627 2.00006 25.9996 7.37263 25.9996 14C25.9996 20.6274 20.627 26 13.9996 26C7.3722 26 1.99963 20.6274 1.99963 14C1.99963 7.37263 7.3722 2.00006 13.9996 2.00006ZM21 13H7.00002C6.44774 13 6.00002 13.4477 6.00002 14C6.00002 14.5523 6.44774 15 7.00002 15H21C21.5523 15 22 14.5523 22 14C22 13.4477 21.5523 13 21 13Z" />{' '}
    </svg>
);

BlockedFilled.defaultProps = {
    xmlns: 'http://www.w3.org/2000/svg',
    fill: 'currentColor',
    width: '28',
    height: '28',
    viewBox: '0 0 28 28',
};

export default BlockedFilled;
