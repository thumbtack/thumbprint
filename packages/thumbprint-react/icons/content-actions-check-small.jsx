import React from 'react';

const ContentActionsCheckSmall = props => (
    <svg {...props}>
        <path
            fillRule="evenodd"
            d="M14.5810763,3.19812449 C14.132085,2.85681117 13.5060971,2.96739243 13.1851033,3.44586905 L8.01720309,11.1408364 L4.52827048,8.83245251 C4.05827956,8.52197434 3.44229146,8.67402358 3.1502971,9.17270255 C2.85930272,9.67138151 3.00329993,10.3274261 3.47229088,10.6379043 L8.55419272,14 L14.8130718,4.68246529 C15.1350656,4.20398867 15.0300676,3.53943781 14.5810763,3.19812449"
        />
    </svg>
);

ContentActionsCheckSmall.defaultProps = {
    xmlns: 'http://www.w3.org/2000/svg',
    fill: 'currentColor',
    width: '18',
    height: '18',
    viewBox: '0 0 18 18',
};

export default ContentActionsCheckSmall;
