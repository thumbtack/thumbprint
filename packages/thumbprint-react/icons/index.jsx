export { default as AlertCheck } from './alert-check.jsx';
export { default as AlertInfo } from './alert-info.jsx';
export { default as AlertWarning } from './alert-warning.jsx';
export { default as BlockedFilled } from './blocked-filled.jsx';
export { default as InfoFilled } from './info-filled.jsx';
export { default as WarningFilled } from './warning-filled.jsx';
