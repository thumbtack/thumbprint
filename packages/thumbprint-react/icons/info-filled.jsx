import React from 'react';

const InfoFilled = props => (
    <svg {...props}>
        <path d="M13.9929 10.4926C13.1655 10.4926 12.4921 9.82131 12.4921 8.99192C12.4921 8.16252 13.1655 7.4912 13.9929 7.4912C14.8213 7.4912 15.4937 8.16252 15.4937 8.99192C15.4937 9.82131 14.8213 10.4926 13.9929 10.4926ZM15.0004 19.0027C15.0004 19.5549 14.5532 20.0031 13.9999 20.0031C13.4466 20.0031 12.9994 19.5549 12.9994 19.0027V12.9998C12.9994 12.4476 13.4466 11.9993 13.9999 11.9993C14.5532 11.9993 15.0004 12.4476 15.0004 12.9998V19.0027ZM22.4883 5.51126C17.8079 0.830039 10.1899 0.829038 5.50955 5.51126C0.830151 10.1915 0.830151 17.8081 5.50955 22.4893C7.84975 24.8304 10.9253 26 13.9999 26C17.0735 26 20.1481 24.8304 22.4883 22.4893C27.1707 17.8081 27.1707 10.1915 22.4883 5.51126Z" />{' '}
    </svg>
);

InfoFilled.defaultProps = {
    xmlns: 'http://www.w3.org/2000/svg',
    fill: 'currentColor',
    width: '28',
    height: '28',
    viewBox: '0 0 28 28',
};

export default InfoFilled;
