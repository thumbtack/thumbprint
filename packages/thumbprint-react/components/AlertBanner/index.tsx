import React from 'react';
import classNames from 'classnames';
import styles from './index.module.scss';
import { BlockedFilled, InfoFilled, WarningFilled } from '../../icons';

interface AlertBannerPropTypes {
    /**
     * Text within the alert
     */
    children: React.ReactNode;
    /**
     * Determines the background color and text color of the component
     */
    theme: 'info' | 'warning' | 'caution';
    /**
     * A selector to hook into the React component for use in automated testing environments.
     */
    dataTestId?: string;
    /**
     * A selector to hook into the React component for use in automated testing environments.
     * @deprecated
     */
    dataTest?: string;
}

const ALERT_ICONS = {
    // Icon names map to the icon name in Thumbprint, not the theme prop
    caution: <WarningFilled className={styles.icon} />,
    info: <InfoFilled className={styles.icon} />,
    warning: <BlockedFilled className={styles.icon} />,
};
export default function AlertBanner({
    children,
    theme,
    dataTestId,
    dataTest,
}: AlertBannerPropTypes): JSX.Element {
    return (
        <div
            className={classNames({
                [styles.root]: true,
                [styles.caution]: theme === 'caution',
                [styles.info]: theme === 'info',
                [styles.warning]: theme === 'warning',
            })}
            data-testid={dataTestId}
            data-test={dataTest}
        >
            {ALERT_ICONS[theme]}
            <div>{children}</div>
        </div>
    );
}
