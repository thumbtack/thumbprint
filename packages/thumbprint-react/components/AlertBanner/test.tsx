import React from 'react';
import { mount } from 'enzyme';
import forEach from 'lodash/forEach';
import AlertBanner from './index';
import { BlockedFilled, InfoFilled, WarningFilled } from '../../icons';

describe('AlertBanner', () => {
    const children = (
        <div data-testid="child">
            This is the child component.
            <a href="/">Click here</a>
        </div>
    );
    const cautionWrapper = mount(
        <AlertBanner theme="caution" dataTest="test">
            {children}
        </AlertBanner>,
    );
    const infoWrapper = mount(
        <AlertBanner theme="info" dataTest="test">
            {children}
        </AlertBanner>,
    );
    const warningWrapper = mount(
        <AlertBanner theme="warning" dataTest="test">
            {children}
        </AlertBanner>,
    );
    const testCases = [cautionWrapper, infoWrapper, warningWrapper];

    test('renders appropriate icons', () => {
        expect(cautionWrapper.find(WarningFilled).length).toBe(1);
        expect(infoWrapper.find(InfoFilled).length).toBe(1);
        expect(warningWrapper.find(BlockedFilled).length).toBe(1);

        expect(cautionWrapper.find('.root.caution').length).toBe(1);
        expect(infoWrapper.find('.root.info').length).toBe(1);
        expect(warningWrapper.find('.root.warning').length).toBe(1);

        forEach(testCases, wrapper => {
            expect(wrapper.find('svg').hasClass('icon')).toBe(true);
        });
    });

    test('renders appropriate text and background color based on theme', () => {
        expect(cautionWrapper.find('.root.caution').length).toBe(1);
        expect(infoWrapper.find('.root.info').length).toBe(1);
        expect(warningWrapper.find('.root.warning').length).toBe(1);
    });

    test('renders children', () => {
        expect(cautionWrapper.find('[data-testid="child"]').length).toBe(1);
    });

    test('appends data-test attribute to root', () => {
        expect(cautionWrapper.find('[data-test="test"]').hasClass('root')).toBe(true);
    });

    test('appends data-testid attribute to root', () => {
        const wrapper = mount(
            <AlertBanner theme="caution" dataTestId="test">
                {children}
            </AlertBanner>,
        );

        expect(wrapper.find('[data-testid="test"]').hasClass('root')).toBe(true);
    });

    test('match snapshot', () => {
        forEach(testCases, wrapper => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});
