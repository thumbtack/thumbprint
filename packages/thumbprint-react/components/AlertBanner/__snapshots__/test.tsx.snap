// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertBanner match snapshot 1`] = `
<AlertBanner
  dataTest="test"
  theme="caution"
>
  <div
    className="root caution"
    data-test="test"
  >
    <WarningFilled
      className="icon"
      fill="currentColor"
      height="28"
      viewBox="0 0 28 28"
      width="28"
      xmlns="http://www.w3.org/2000/svg"
    >
      <svg
        className="icon"
        fill="currentColor"
        height="28"
        viewBox="0 0 28 28"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13.9985 21.5C13.1695 21.5 12.4985 20.829 12.4985 20C12.4985 19.171 13.1695 18.5 13.9985 18.5C14.8275 18.5 15.4985 19.171 15.4985 20C15.4985 20.829 14.8275 21.5 13.9985 21.5ZM12.9995 10C12.9995 9.448 13.4475 9 13.9995 9C14.5525 9 14.9995 9.448 14.9995 10V16C14.9995 16.552 14.5525 17 13.9995 17C13.4475 17 12.9995 16.552 12.9995 16V10ZM25.4195 18.502L17.7805 5.187C16.9955 3.817 15.5815 3 13.9995 3C12.4175 3 11.0045 3.817 10.2195 5.187L2.58054 18.502C1.80154 19.858 1.80554 21.476 2.58854 22.829C3.37554 24.188 4.78654 25 6.36054 25H21.6395C23.2135 25 24.6235 24.188 25.4115 22.829C26.1945 21.476 26.1985 19.858 25.4195 18.502Z"
        />
         
      </svg>
    </WarningFilled>
    <div>
      <div
        data-testid="child"
      >
        This is the child component.
        <a
          href="/"
        >
          Click here
        </a>
      </div>
    </div>
  </div>
</AlertBanner>
`;

exports[`AlertBanner match snapshot 2`] = `
<AlertBanner
  dataTest="test"
  theme="info"
>
  <div
    className="root info"
    data-test="test"
  >
    <InfoFilled
      className="icon"
      fill="currentColor"
      height="28"
      viewBox="0 0 28 28"
      width="28"
      xmlns="http://www.w3.org/2000/svg"
    >
      <svg
        className="icon"
        fill="currentColor"
        height="28"
        viewBox="0 0 28 28"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13.9929 10.4926C13.1655 10.4926 12.4921 9.82131 12.4921 8.99192C12.4921 8.16252 13.1655 7.4912 13.9929 7.4912C14.8213 7.4912 15.4937 8.16252 15.4937 8.99192C15.4937 9.82131 14.8213 10.4926 13.9929 10.4926ZM15.0004 19.0027C15.0004 19.5549 14.5532 20.0031 13.9999 20.0031C13.4466 20.0031 12.9994 19.5549 12.9994 19.0027V12.9998C12.9994 12.4476 13.4466 11.9993 13.9999 11.9993C14.5532 11.9993 15.0004 12.4476 15.0004 12.9998V19.0027ZM22.4883 5.51126C17.8079 0.830039 10.1899 0.829038 5.50955 5.51126C0.830151 10.1915 0.830151 17.8081 5.50955 22.4893C7.84975 24.8304 10.9253 26 13.9999 26C17.0735 26 20.1481 24.8304 22.4883 22.4893C27.1707 17.8081 27.1707 10.1915 22.4883 5.51126Z"
        />
         
      </svg>
    </InfoFilled>
    <div>
      <div
        data-testid="child"
      >
        This is the child component.
        <a
          href="/"
        >
          Click here
        </a>
      </div>
    </div>
  </div>
</AlertBanner>
`;

exports[`AlertBanner match snapshot 3`] = `
<AlertBanner
  dataTest="test"
  theme="warning"
>
  <div
    className="root warning"
    data-test="test"
  >
    <BlockedFilled
      className="icon"
      fill="currentColor"
      height="28"
      viewBox="0 0 28 28"
      width="28"
      xmlns="http://www.w3.org/2000/svg"
    >
      <svg
        className="icon"
        fill="currentColor"
        height="28"
        viewBox="0 0 28 28"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13.9996 2.00006C20.627 2.00006 25.9996 7.37263 25.9996 14C25.9996 20.6274 20.627 26 13.9996 26C7.3722 26 1.99963 20.6274 1.99963 14C1.99963 7.37263 7.3722 2.00006 13.9996 2.00006ZM21 13H7.00002C6.44774 13 6.00002 13.4477 6.00002 14C6.00002 14.5523 6.44774 15 7.00002 15H21C21.5523 15 22 14.5523 22 14C22 13.4477 21.5523 13 21 13Z"
        />
         
      </svg>
    </BlockedFilled>
    <div>
      <div
        data-testid="child"
      >
        This is the child component.
        <a
          href="/"
        >
          Click here
        </a>
      </div>
    </div>
  </div>
</AlertBanner>
`;
